patina_dxe_core/src/lib.rs

```diff
@@ -649,4 +659,19 @@ mod tests {
 
         assert!(!<TestPlatform as MemoryInfo>::prioritize_32_bit_memory());
     }
+
+    #[test]
+    #[should_panic = "BDS arch protocol should be found and should never return."]
+    fn test_mock_call_bds() {
```

Comment:
This needs way more tests. Basically a full enumerated test of scenarios where `status_code_protocol` does or does not exist, and if it does exist, is or is not null and where `bds` does or does not exist, and if it does exist, is or is not null.